lib/openstack_query/runners/runner_wrapper.py

```diff
@@ -34,80 +34,112 @@ def __init__(self, marker_prop_func: PropFunc, connection_cls=OpenstackConnectio
     def run(
         self,
         cloud_account: str,
-        client_side_filter_func: Optional[ClientSideFilterFunc] = None,
+        client_side_filters: Optional[ClientSideFilters] = None,
         server_side_filters: Optional[ServerSideFilters] = None,
         from_subset: Optional[List[Any]] = None,
         **kwargs,
     ) -> List[OpenstackResourceObj]:
         """
         Public method that runs the query by querying openstacksdk and then applying a filter function.
         :param cloud_account: An string for the account from the clouds configuration to use
-        :param client_side_filter_func: An Optional function that we can use to limit the results after querying
-        openstacksdk
-        :param server_side_filters: An Optional set of filter kwargs to limit the results by when querying openstacksdk
+        :param client_side_filters: An Optional list of filter functions to run locally that we can use to limit the
+        results after querying openstacksdk
+        :param server_side_filters: An Optional list of filter kwargs to limit the results by when querying openstacksdk
         :param from_subset: A subset of openstack resources to run query on instead of querying openstacksdk
         :param kwargs: An extra set of kwargs to pass to internal _run_query method that changes what/how the
         openstacksdk query is run
             - valid kwargs to _run_query is specific to the runner object - see docstrings for _run_query() on the
             runner of interest.
         """
+
+        if from_subset and server_side_filters:
+            logger.error(
+                "Error: Received server-side filters - suggesting that the query requires getting results via "
+                "openstacksdk, but also openstack objects directly - suggesting that we want to filter a set already "
+                "given. Not sure what to do - aborting."
+            )
+            raise RuntimeError(
+                "Ambiguous Query: received both server-side filters and values passed directly"
+            )
```

Comment:
We should extract this to something like `self._validate_params` to prevent the run growing

Reply:
for a future PR